fluent/fluent.go

```diff
@@ -204,6 +210,8 @@ func (f *Fluent) close() (err error) {
 
 // connect establishes a new connection using the specified transport.
 func (f *Fluent) connect() (err error) {
+	f.mu.Lock()
+	defer f.mu.Unlock()
```

Comment:
`Dial` may take long time.  This lock shouldn't prevent write from other goroutines.